src/btree/bt_cursor.c

```diff
@@ -1360,9 +1365,14 @@ __cursor_chain_exceeded(WT_CURSOR_BTREE *cbt)
 	    page->modify != NULL && page->modify->mod_row_update != NULL)
 		upd = page->modify->mod_row_update[cbt->slot];
 
-	for (i = 0; upd != NULL; ++i, upd = upd->next) {
+	for (i = 0, upd_size = 0; upd != NULL; ++i, upd = upd->next) {
 		if (WT_UPDATE_DATA_VALUE(upd))
 			return (false);
+		upd_size += WT_UPDATE_MEMSIZE(upd);
+		if (!__wt_txn_upd_visible_all(session, upd))
+			continue;
```

Comment:
We should explain that when we have to maintain history, creating extra copies of a large document multiplies the cache pressure, so we avoid it.